### PR TITLE
Serverside Test runner

### PR DESCRIPTION
**Pull request by shreymahendru**
The script goes through all the tests in the ``testDAO`` and runs them while logging appropriate messages. 
The can be run from through command line using following arg
`` -Dfoam.main=testRunnerScript ``

### CODE REVIEW

**Review by jlhughes — APPROVED** [review]

File: src/foam/nanos/script/TestRunnerScript.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      name: 'RESET_COLOR',
+      type: 'String',
+      value: '\u001B[0m'
+    }
+  ],
+  properties: [
+    {
+      name: 'failedTests',
+      class: 'Int',
+      value: 0
```
Comment (kgrgreer) [inline src/foam/nanos/script/TestRunnerScript.js]: 0 is already the default value

File: src/foam/nanos/script/TestRunnerScript.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    },
+    {
+      name: 'passedTests',
+      class: 'Int',
+      value: 0
```
Comment (kgrgreer) [inline src/foam/nanos/script/TestRunnerScript.js]: 0 is already the default value

File: src/foam/nanos/script/TestRunnerScript.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        LogLevelFilterLogger loggerFilter = (LogLevelFilterLogger) x.get("logger");
+        loggerFilter.setLogInfo(false);
+        loggerFilter.setLogDebug(false);
+        loggerFilter.setLogWarning(false);
+
+        DAO testDAO = (DAO) x.get("testDAO");
+        ArraySink tests = (ArraySink) testDAO.select(new ArraySink());
+        List testArray = tests.getArray();
+
+        for(int i = 0; i < testArray.size(); i ++) {
```
Comment (kgrgreer) [inline src/foam/nanos/script/TestRunnerScript.js]: format:
    for ( int i = 0 ; i < testArray.size() ; i++ ) {

File: src/foam/nanos/script/TestRunnerScript.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+          Test test = (Test) testArray.get(i);
+          test = (Test) test.fclone();
+          if( test.getServer() ) {
```
Comment (kgrgreer) [inline src/foam/nanos/script/TestRunnerScript.js]: space after if

File: src/foam/nanos/script/TestRunnerScript.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+          name: 'x', javaType: 'foam.core.X'
+        },
+        {
+          name: 'test', javaType: 'Test'
+        }
+      ],
+      javaCode: `
+        printBold(test.getId());
+        try {
+            test.runScript(x);
```
Comment (kgrgreer) [inline src/foam/nanos/script/TestRunnerScript.js]: indent only 2 spaces

File: src/foam/nanos/script/TestRunnerScript.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+          }
+          else {
+            if ( ! SafetyUtil.isEmpty(output) ) {
+              System.out.println(output);
+            }
+          }
+        }
+      `
+    }
+
```
Comment (kgrgreer) [inline src/foam/nanos/script/TestRunnerScript.js]: remove blank line

File: src/foam/nanos/script/TestRunnerScript.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+  ]
+
```
Comment (kgrgreer) [inline src/foam/nanos/script/TestRunnerScript.js]: remove blank line

**Pull request merged**